src/test_utils/mongo/mod.rs

```diff
@@ -523,33 +274,118 @@ impl<'a> MongoFuzzer<'a> {
     }
 }
 
+/// Builder for constructing transactions with custom values.
+pub struct TransactionBuilder {
+    /// The type of transaction to construct.
+    tx_type: Option<TxType>,
```

Comment:
can't this just be a TxType? and then if we don't call `tx_type` it's just the default tx type.

Reply:
Don't think because I use this as an `Option` to ensure that if `None` I generate an arbitrary transaction and if `Some` then I generate a hardcoded transaction for the specific needs of certain tests (for example those where we verify signatures) .

In the `None` case, I would like the transaction type to be completely random.